### DISCUSSION TITLE
Is there a way to safely chain zrepl replications ?

### DISCUSSION THREAD

**Discussion by otopetrik** [zrepl/zrepl#603] [Q&A]
Hello,

I am looking for a way to perform periodic backup of "laptop SSD" datasets using a chain of replications:
1. laptop SSD to laptop HDD
2. laptop HDD to home NAS
3. home NAS to remote NAS

It seems that zrepl cannot be used to do that.

- Each replication "jump" lengthens the full path to datasets ([#253](https://github.com/zrepl/zrepl/issues/253)), that would make paths on remote NAS needlessly long.

- Using target dataset of first replicating job as source dataset for second replication does not seem to work ([#490](https://github.com/zrepl/zrepl/issues/490)).

Documentation mentions only "Fan-in" and "Fan-out" setups, so it looks like zrepl replications cannot be safely chained (there is no safe workaround for #490). Is that correct ?

Thanks.

**Comment by problame** [comment]
Yes, that is correct. I guess fixing #490  and #253  would enable the chaining use case.

But honestly, instead of doing 3 levels of ZFS replication, I'd rather try to diversify. E.g., use borgbackup or restic to create another type of backup that doesn't rely on zfs send/recv.

**Reply by kpfleming** [reply]
In my situation that's not an option, as I want to put copies of encrypted filesystems onto 'untrusted' media (or machines), so that means they have to be ZFS raw transfers.

**Reply by problame** [reply]
restic or borgbackup support encryption, and I think it's actually the default
